src/js/components/Form/Form.js

```diff
@@ -571,7 +571,10 @@ const Form = forwardRef(
           }
         }}
       >
-        <FormContext.Provider value={{ useFormField, useFormInput }}>
+        <FormContext.Provider
+          // eslint-disable-next-line react/jsx-no-constructed-context-values
```

Comment:
should we add a TODO comment?

Reply:
This one is a quite a bit more involved. It also is linked to some potential performance wins which we have documented in an existing issue.

Reply:
I think changing it will improve performance.

Is this one really a lot more involved? Can we create the FormContext value via `useMemo`?

Reply:
In theory, I do think changing it to comply with the eslint rule would likely lead to performance gains, but not enough to address other performance issues it has.

Today, if anything changes in the form's context, the entire Form and all children (FormFields, inputs) are all re-rendered. This is because the form's value and its validation status are important to all the field/inputs.

As example, a TextInput which is validated onChange will affect the value held by the entire form, will trigger form validation, and thus the form's overall context. All subcomponents dependent on that context are then re-rendered due to a single key stroke. Add lots of fields to the form, which are all getting re-rendered with every keystroke eventually becomes problematic.

Ideally, only the form and input being validated is re-rendered. I think to solve for this will require rearchitecting, possibly by splitting up the contexts. Refactoring the hooks driving the context will be quite invasive and I am not fully confident that it would address the "all children re-rendering" on each onChange.

Here's the [issue and some related discussion](https://github.com/grommet/grommet/issues/4559#issuecomment-727025196).

Reply:
@ericsoderberghp - I will take a look again, but I do think it is more invasive. I'll double check.

Reply:
I did revisit this and was able to memoize the FormContext value here. Additionally, I did some testing with React DevTools Profiler and found that:
1) Reduced the number of commits from 18 to 6.
2) Eliminated the re-rendering of all child inputs after the onChange event. Reduced from 27ms to 2ms.
3) Reduced total render time for equivalent form update from 40ms to 12ms.

[Gist with Profiler Data](https://gist.github.com/halocline/36060bbda2f1ae6ece6b9d35826c0d5f) - Upload to React Dev Tools Profiler - Generated from FireFox

**Before memo**
![Screen Shot 2021-12-17 at 2 51 58 PM](https://user-images.githubusercontent.com/1756948/146612710-ed2c0091-05b6-4a0b-b00d-55b10294ee71.png)

**After memo**
![Screen Shot 2021-12-17 at 2 52 33 PM](https://user-images.githubusercontent.com/1756948/146612713-e5e9b175-59c5-40d8-ad42-ce17d1e4c6c9.png)